docs/modules/ROOT/pages/udc.adoc

```diff
@@ -0,0 +1,228 @@
+
+The UDC deployment should also set `deploy_from_zero=TRUE` in the `deploy` syscall.
+In so doing, this creates a deterministic address that can be used across all instances of StarkNet.
+
+== Usage
+
+The Universal Deployer Contract offers two deployment types: reserving the address space for the deployer and reproducible deployments.
+The `unique` boolean parameter ultimately determines the type of deployment.
+
+CAUTION: When deploying a contract that uses `get_caller_address` in the constructor calldata, remember that the UDC (and not the account) deploys that contract.
+Therefore, querying `get_caller_address` in a contract's constructor will return the UDC's address and *not* the account's address.
+
+
+=== Reserved Address Space for Deployer
+
+Reserving the address space for the deployer deploys a contract unique to the deployer.
+This prevents squatting on addresses.
+
+Achieving this type of deployment necessitates that the deployer sets `unique` to `TRUE` in the `deployContract` call.
+Under the hood, the function call leverages the deployer's address and creates a hashchain by hashing the deployer's address with the given salt.
+
+To deploy a unique contract address:
+
+[,python]
+----
+signer = MockSigner(PRIVATE_KEY)
+
+async def deploy_contract():
+    # Declare the contract and get the class hash before deployment
+    class_hash, _ = await signer.declare_class(account, "Ownable")
+
+    # Set up deployment parameters
+    salt = 123456  # random number
+    unique = TRUE
+    calldata = [account.contract_address]
+    params = [class_hash, salt, unique, len(calldata), *calldata]
+
+    # Send a "deployContract" transaction to the deployer contract
+    await account.send_transaction(
+        account,
+        deployer_address,
+        "deployContract",
+        params
+    )
+----
+
+
+=== Reproducible Deployments
+
+Reproducible deployments create contract addresses independent of the deployer and the UDC instance.
+Instead, only the class hash, salt, and constructor arguments determine the address.
+This type of deployment enables redeployments of accounts and known systems across multiple networks.
+To deploy a reproducible deployment, set `unique` to `FALSE`.
+
+[,python]
+----
+signer = MockSigner(PRIVATE_KEY)
+
+async def deploy_contract():
+    # Declare the contract and get the class hash before deployment
+    class_hash, _ = await signer.declare_class(account, "Ownable")
+
+    # Set up deployment parameters
+    salt = 123456  # random number
+    unique = FALSE
+    calldata = [account.contract_address]
+    params = [class_hash, salt, unique, len(calldata), *calldata]
+
+    # Send a "deployContract" transaction to the deployer contract
+    await account.send_transaction(
+        account,
+        deployer_address,
+        "deployContract",
+        params
+    )
+----
+
+== Calculating Counterfactual Addresses
```

Comment:
```suggestion
== Calculating counterfactual addresses
```

Reply:
why is this section relevant to the UDC? i think i prefer to keep it because is loosely related to deployments and might come handy, but otherwise i see no relation with the UDC

Reply:
I was thinking that if a user wanted info on calculating addresses when deploying through the UDC, this would be the logical place to look. I'm not sure if there's a better home for it ATM.

I do agree it's not directly relevant to the UDC. Your call if you'd like to keep it here